packages/@aws-cdk/aws-codedeploy/lib/server/load-balancer.ts

```diff
@@ -0,0 +1,90 @@
+import elb = require('@aws-cdk/aws-elasticloadbalancing');
+import elbv2 = require('@aws-cdk/aws-elasticloadbalancingv2');
+
+/**
+ * The generations of AWS load balancing solutions.
+ */
+export enum LoadBalancerGeneration {
+  /**
+   * The first generation (ELB Classic).
+   */
+  FIRST = 0,
+
+  /**
+   * The second generation (ALB and NLB).
+   */
+  SECOND = 1
+}
+
+/**
+ * An interface of an abstract load balancer, as needed by CodeDeploy.
+ * Create instances using the static factory methods:
+ * {@link #classic}, {@link #alb} and {@link #nlb}.
+ */
+export abstract class LoadBalancer {
+  /**
+   * Creates a new CodeDeploy load balancer from a Classic ELB Load Balancer.
+   *
+   * @param loadBalancer a classic ELB Load Balancer
+   */
+  public static classic(loadBalancer: elb.LoadBalancer): LoadBalancer {
+    return new ClassicLoadBalancer(loadBalancer);
+  }
+
+  /**
+   * Creates a new CodeDeploy load balancer from an Application Load Balancer Target Group.
+   *
+   * @param albTargetGroup an ALB Target Group
+   */
+  public static alb(albTargetGroup: elbv2.ApplicationTargetGroup): LoadBalancer {
```

Comment:
I think I'd prefer a longer name here.

Also, I have no idea how this makes sense, to create an ALB "from" a target group. But oh well I suppose?

Reply:
How about `application`?

Reply:
Changed to `application`.